wordpress-mobile/WordPress-iOS
# Add the Kanvas editor for creating a story

Adds the Kanvas editor for Story creation, does not hook up editing or other advanced functionality.

### Overview

The `StoryEditor` is a new type of Editor which displays the Kanvas `CameraController`. It handles the default settings, setting up the delegate (`CameraHandler`) and handling blog/post types for WPiOS. The `PublishingEditor` protocol splits out common publishing logic so it can be shared amongst the editors.

### Testing

* Open to the Blog Detail page
* Tap the FAB
* Select the "Story post" option
* Ensure that the Kanvas Editor shows up and basic photo, video, and media import options work.
* Hit the blue confirm button from the edit screen.
* Confirm that the Prepublishing sheet is shown.

Note that publishing will _NOT_ work with these changes. This is focused on pulling in Kanvas and properly displaying the interfaces. A subsequent PR will focus on publishing + Gutenberg editing.

https://user-images.githubusercontent.com/3250/108170984-15627700-70b8-11eb-969d-8921250fed9b.mp4

PR submission checklist:
